File: packages/meterbus-wired-datalink-python/examples/decode_exact.py
"""Decode one complete frame, inspect its fields, and encode it again."""

from meterbus_wired_datalink import Frame

frame = Frame.decode(bytes.fromhex("105b015c16"))

assert frame.kind == "short"
assert frame.control_byte == 0x5B
assert frame.address == 1
assert frame.encode() == bytes.fromhex("105b015c16")

File: packages/meterbus-wired-datalink-node/main/examples/decode-exact.ts
// Decode one complete frame, inspect its fields, and encode it again.
import assert from "node:assert/strict";
import { Frame } from "@meterbus/wired-datalink-node";

const frame = Frame.decode(Uint8Array.of(0x10, 0x5b, 0x01, 0x5c, 0x16));

assert.equal(frame.kind, "short");
assert.equal(frame.controlByte, 0x5b);
assert.equal(frame.address, 1);
assert.deepEqual(frame.encode(), Uint8Array.of(0x10, 0x5b, 0x01, 0x5c, 0x16));
